openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/DatabaseSchemaPage.component.tsx

```diff
@@ -531,6 +576,29 @@ const DatabaseSchemaPage: FunctionComponent = () => {
     pageSize,
   ]);
 
+  // set pagesize and current page on page reload from history state if present
+  useEffect(() => {
+    const cursorState = pagingCursor;
+    if (cursorState?.cursorData?.cursorType) {
+      handlePageSizeChange(cursorState?.pageSize as number);
+      handlePageChange(
+        cursorState.currentPage as number,
+        cursorState?.cursorData,
+        cursorState?.pageSize
+      );
+    } else {
+      handlePageChange(
+        currentPage,
+        {
+          cursorType: null,
+          cursorValue: null,
+        },
+        cursorState?.pageSize ?? pageSize
+      );
+      handlePageSizeChange(cursorState?.pageSize ?? pageSize);
+    }
+  }, []);
```

Comment:
This should be handled by usePaging instead

Reply:
shifted this inside usePaging